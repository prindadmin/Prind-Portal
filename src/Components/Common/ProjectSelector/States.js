export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
