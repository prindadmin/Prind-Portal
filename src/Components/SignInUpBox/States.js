export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
