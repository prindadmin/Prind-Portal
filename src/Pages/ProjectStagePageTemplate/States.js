export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
