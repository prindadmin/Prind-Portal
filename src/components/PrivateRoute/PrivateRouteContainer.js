import { connect } from 'react-redux'

import PrivateRouteComponent from './PrivateRoute'

import * as userReducer from '../../Reducers/userReducer'
import * as projectsReducer from '../../Reducers/projectsReducer'

const mapStatetoProps = state => {
  return {
    auth: state.auth,
    user: state.user,
    projects: state.projects,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    storeRoute: (route) => {
      dispatch(userReducer.storeRoute(route))
    },
    updateChosenProject: (idToken, project, resolve) => {
      console.log("getting project details")
      dispatch(projectsReducer.updateChosenProject(idToken, project, resolve))
    },
  }
}

export default connect(mapStatetoProps, mapDispatchToProps)(PrivateRouteComponent)
