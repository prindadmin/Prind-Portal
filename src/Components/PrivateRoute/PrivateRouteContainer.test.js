import { Provider, connect } from 'react-redux'
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import {
  HashRouter as Router,
  Route,
} from "react-router-dom";

import * as Component from './PrivateRouteContainer'
import * as States from '../../States'

import SubComponent from '../Error404'

// https://www.robinwieruch.de/react-connected-component-test

const mockStore = configureStore([]);

var component;

beforeEach(() => {
  const store = mockStore({
    auth: {
      info: {},
      error: {},
      isSignedIn: States.AUTH_UNKNOWN,
      isConfirmed: States.AUTH_UNKNOWN,
      hasSignedUp: States.AUTH_UNKNOWN,
      hasSentCode: States.AUTH_UNKNOWN,
      hasChangedPassword: States.AUTH_UNKNOWN,
      passwordResetRequired: States.AUTH_UNKNOWN
    },
    user: {
      fetching: false,
      currentRoute: '/profile',
      details: {},
    }
  });

  const mockStoreRoute = jest.fn()
  const props = {
    auth: {
      isSignedIn: States.AUTH_UNKNOWN
    },
    path: '/documents',
    location: {
      pathname: '/documents'
    },
    user: {
      currentRoute: '/documents'
    },
    component: <SubComponent />,
    storeRoute: mockStoreRoute
  };

  component = mount(
    <Provider store={store}>
      <Router>
        <Component.default {...props} />
      </Router>
    </Provider>
  );
});


it('should render with given state from Redux store', () => {
  expect(component).toMatchSnapshot();
});
