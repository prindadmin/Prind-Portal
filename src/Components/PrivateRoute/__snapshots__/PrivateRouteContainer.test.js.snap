// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with given state from Redux store 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "REPLACE",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(PrivateRoute)
        auth={
          Object {
            "isSignedIn": "AUTH_UNKNOWN",
          }
        }
        component={<Error404 />}
        location={
          Object {
            "pathname": "/documents",
          }
        }
        path="/documents"
        storeRoute={[MockFunction]}
        user={
          Object {
            "currentRoute": "/documents",
          }
        }
      >
        <PrivateRoute
          auth={
            Object {
              "error": Object {},
              "hasChangedPassword": "AUTH_UNKNOWN",
              "hasSentCode": "AUTH_UNKNOWN",
              "hasSignedUp": "AUTH_UNKNOWN",
              "info": Object {},
              "isConfirmed": "AUTH_UNKNOWN",
              "isSignedIn": "AUTH_UNKNOWN",
              "passwordResetRequired": "AUTH_UNKNOWN",
            }
          }
          component={<Error404 />}
          location={
            Object {
              "pathname": "/documents",
            }
          }
          path="/documents"
          storeRoute={[Function]}
          updateChosenProject={[Function]}
          user={
            Object {
              "currentRoute": "/profile",
              "details": Object {},
              "fetching": false,
            }
          }
        >
          <Route
            auth={
              Object {
                "error": Object {},
                "hasChangedPassword": "AUTH_UNKNOWN",
                "hasSentCode": "AUTH_UNKNOWN",
                "hasSignedUp": "AUTH_UNKNOWN",
                "info": Object {},
                "isConfirmed": "AUTH_UNKNOWN",
                "isSignedIn": "AUTH_UNKNOWN",
                "passwordResetRequired": "AUTH_UNKNOWN",
              }
            }
            location={
              Object {
                "pathname": "/documents",
              }
            }
            path="/documents"
            render={[Function]}
            storeRoute={[Function]}
            updateChosenProject={[Function]}
            user={
              Object {
                "currentRoute": "/profile",
                "details": Object {},
                "fetching": false,
              }
            }
          >
            <Redirect
              to="/"
            >
              <Lifecycle
                onMount={[Function]}
                onUpdate={[Function]}
                to="/"
              />
            </Redirect>
          </Route>
        </PrivateRoute>
      </Connect(PrivateRoute)>
    </Router>
  </HashRouter>
</Provider>
`;
